rfcs/react-components/convergence/long-term-support-plan.md

```diff
@@ -0,0 +1,122 @@
+
+We should announce an end-of-life to v7. Proposed: June 30th, 2024. We would lock the branch from any changes, disable and archive any build pipelines, and replace public documentation sites with a redirect to a notice that v7 is retired.
+
+Any partners wanting to continue using v7, should take a snapshot of the code and integrate it into their own repositories and build it.
+
+#### Separate v0, v8 and v9 branches
+
+v7 is already in its own branch. v0, v8, and v9 are in the master branch which causes some confusion for developers.
+
+1. Create a v8 branch from master
+2. Create a northstar-v0 branch from master
+3. Remove all v9 code from the northstar-v0, and v8 branches
+4. Remove all v0 and v8 code from the master branch
+5. Update v8 components to take npm dependencies on compat components
+6. Update any migration components to take npm dependencies on v8 components
+7. Update v0, v8 builds to publish only v0, v8 components
+8. Update v9 build to publish only v9 components
+9. Update README.MD in each branch to cover its version and point to the other branches
+10. Update ADO pipelines to handle each different branch (triggers, builds, releases)
+11. Update ADO documentation for releasing each version
+
+- Keep cross-version compat components in master (v9).
+- Keep migration components in master (v9).
+
+This plan requires communicating to partners that own their own v0 and v8 based packages in the master branch that they will need to start working on them in the new branches.
+
+There should be a cut-over date where branches are locked for changes and then the steps taken (scripts?) to make the change followed by the branches being unlocked. It would be good to time
+the cutover to a time of low activity for all teams globally.
+
+#### Flatten master branch
+
+With northstar-v0 and v8 no longer in the master branch, we can reduce the depth of the tree by bringing components within the react-components folder up one level.
+
+#### Sunset v8
+
+After retiring v7, announce v8 is in sunset mode. Only critical bugs will be fixed. We will no longer be moving React forward in v8. Set a retirement date for one year out.
+
+#### Define team tenets for active versions
+
+The Fluent team size doesn't allow for supporting more than 2 versions of Fluent. We should consider policy where we don't start active development on a new major version until the oldest version is retired (or at least in sunset). We can have a next branch and do lots of experimentation there, but there would be no support for this branch.
+
+Branch renames make it possible to have an easier current verson change if we do next version development it its own branch. We can rename master to vX and then rename next to master.
+
+Note: We should also consider renaming master to main.
+
+### Pros and Cons
+
+- 👍 v7 is already isolated in own branch, requiring no changes and will be familiar to partners
+- 👍 branches provide independent policy allowing for easier management
```

Comment:
can you please elaborate on this one ?

Reply:
It is easy in GitHub to set up branch protection policy, CI build policy, etc per branch vs having to specify a set of subfolder paths.